### ISSUE TITLE
RemoteError object is not iterable

### ISSUE THREAD

**Issue by jakirkham** [ipython/ipyparallel#162]
Getting an interesting error in ipyparallel 5.1.1 on python 3.5 in the process of handling another error. Traceback shown below.

```
/groups/dudman/home/kirkhamj/miniconda/envs/nanshenv3/lib/python3.5/site-packages/ipyparallel/client/asyncresult.py in __iter__(self)
    670     def __iter__(self):
    671         it = self._ordered_iter if self.ordered else self._unordered_iter
--> 672         for r in it():
    673             yield r
    674 

/groups/dudman/home/kirkhamj/miniconda/envs/nanshenv3/lib/python3.5/site-packages/ipyparallel/client/asyncresult.py in _ordered_iter(self)
    699             for child in self._children:
    700                 self._wait_for_child(child, evt=evt)
--> 701                 for r in self._yield_child_results(child):
    702                     yield r
    703         else:

/groups/dudman/home/kirkhamj/miniconda/envs/nanshenv3/lib/python3.5/site-packages/ipyparallel/client/asyncresult.py in _yield_child_results(self, child)
    679         """
    680         rlist = child.result()
--> 681         error.collect_exceptions(rlist, self._fname)
    682         try:
    683             for r in rlist:

/groups/dudman/home/kirkhamj/miniconda/envs/nanshenv3/lib/python3.5/site-packages/ipyparallel/error.py in collect_exceptions(rdict_or_list, method)
    210     else:
    211         rlist = rdict_or_list
--> 212     for r in rlist:
    213         if isinstance(r, RemoteError):
    214             en, ev, etb, ei = r.ename, r.evalue, r.traceback, r.engine_info

TypeError: 'RemoteError' object is not iterable
```

**Comment by minrk** [comment]
Same as #163, I thought this was fixed in 5.1.1. Can you reinstall and verify that it's not 5.1.0 where you are seeing this? If you can reproduce it in 5.1.1, can you include a sample that reproduces it?

**Comment by jakirkham** [comment]
> Same as #163, I thought this was fixed in 5.1.1.

Yeah, I noticed a reference to the former was in the release info. Didn't know about this one though.

> Can you reinstall and verify that it's not 5.1.0 where you are seeing this?

Interestingly, I had to create a totally new Miniconda install for unrelated reasons. So, it started out with 5.1.1 from conda-forge.

> If you can reproduce it in 5.1.1, can you include a sample that reproduces it?

Sure, I'll work on some simple examples. Not sure if it will happen this week, but should be able to take a closer look next week. Feel free to politely remind if I haven't done it by then.
